# **DATA ASSESSING**

In [13]:
import numpy as np
import pandas as pd

In [14]:
df = pd.read_csv('Ecommerce.csv')

In [15]:
df.head()

customer_id                          order_id  \
0  9ef432eb6251297304e76186b10a928d  e481f51cbdc54678b7cc49136f2d6af7   
1  9ef432eb6251297304e76186b10a928d  e481f51cbdc54678b7cc49136f2d6af7   
2  9ef432eb6251297304e76186b10a928d  e481f51cbdc54678b7cc49136f2d6af7   
3  b0830fb4747a6c6d20dea0b8c802d7ef  53cdb2fc8bc7dce0b6741e2150273451   
4  41ce2a54c0b03bf3443c3d931a367089  47770eb9100c2d0c44946d9cf07ec65d   

   order_item_id                customer_unique_id customer_state  \
0              1  7c396fd4830fd04220f754e42b4e5bff             SP   
1              1  7c396fd4830fd04220f754e42b4e5bff             SP   
2              1  7c396fd4830fd04220f754e42b4e5bff             SP   
3              1  af07308b275d755c9edb36a90c618231             BA   
4              1  3a653a41f6f9fc3d2a113cf8398680e8             GO   

  customer_city                        product_id  product_category_name  \
0     sao paulo  87285b34884572647811a353c7ac498a  utilidades_domesticas   
1     sao paulo  87285b34884572647811a353c7ac498a  utilidades_domesticas   
2     sao paulo  87285b34884572647811a353c7ac498a  utilidades_domesticas   
3     barreiras  595fac2a385ac33a80bd5114aec74eb8             perfumaria   
4    vianopolis  aa4383b373c6aca5d8797843e5594415             automotivo   

    price  product_name_lenght  ...  order_delivered_customer_date  \
0   29.99                 40.0  ...            2017-10-10 21:25:13   
1   29.99                 40.0  ...            2017-10-10 21:25:13   
2   29.99                 40.0  ...            2017-10-10 21:25:13   
3  118.70                 29.0  ...            2018-08-07 15:27:45   
4  159.90                 46.0  ...            2018-08-17 18:06:29   

   order_status  order_estimated_delivery_date  payment_type  \
0     delivered            2017-10-18 00:00:00   credit_card   
1     delivered            2017-10-18 00:00:00       voucher   
2     delivered            2017-10-18 00:00:00       voucher   
3     delivered            2018-08-13 00:00:00        boleto   
4     delivered            2018-09-04 00:00:00   credit_card   

   payment_installments  payment_value payment_sequential freight_value  \
0                     1          18.12                  1          8.72   
1                     1           2.00                  3          8.72   
2                     1          18.59                  2          8.72   
3                     1         141.46                  1         22.76   
4                     3         179.12                  1         19.22   

  customer_zip_code_prefix  shipping_limit_date  
0                     3149  2017-10-06 11:07:15  
1                     3149  2017-10-06 11:07:15  
2                     3149  2017-10-06 11:07:15  
3                    47813  2018-07-30 03:24:27  
4                    75265  2018-08-13 08:55:23  

[5 rows x 40 columns]

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 115609 entries, 0 to 115608
Data columns (total 40 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   customer_id                    115609 non-null  object 
 1   order_id                       115609 non-null  object 
 2   order_item_id                  115609 non-null  int64  
 3   customer_unique_id             115609 non-null  object 
 4   customer_state                 115609 non-null  object 
 5   customer_city                  115609 non-null  object 
 6   product_id                     115609 non-null  object 
 7   product_category_name          115609 non-null  object 
 8   price                          115609 non-null  float64
 9   product_name_lenght            115609 non-null  float64
 10  product_description_lenght     115609 non-null  float64
 11  product_photos_qty             115609 non-null  float64
 12  product_weight_g              

###  Columns needs to be Dropped

- customer_id  
- product_category_name  
- review_id  
- review_comment_title  
- review_comment_message  
- review_creation_date  
- review_answer_timestamp  
- payment_sequential  
- payment_installments   
- shipping_limit_date  
- seller_zip_code_prefix  
- customer_zip_code_prefix  
- product_name_lenght  
- product_description_lenght  
- product_photos_qty  
- seller_city  
- customer_city  
- order_approved_at  
- order_delivered_carrier_date
- order_estimated_delivery_date



In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 115609 entries, 0 to 115608
Data columns (total 40 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   customer_id                    115609 non-null  object 
 1   order_id                       115609 non-null  object 
 2   order_item_id                  115609 non-null  int64  
 3   customer_unique_id             115609 non-null  object 
 4   customer_state                 115609 non-null  object 
 5   customer_city                  115609 non-null  object 
 6   product_id                     115609 non-null  object 
 7   product_category_name          115609 non-null  object 
 8   price                          115609 non-null  float64
 9   product_name_lenght            115609 non-null  float64
 10  product_description_lenght     115609 non-null  float64
 11  product_photos_qty             115609 non-null  float64
 12  product_weight_g              

### 🧾 Column Renaming
- customer_unique_id -> customer_id
- freight value -> delivery_charge
- product_category_name_english -> category
- review_score -> rating
- order_purchase_timestamp -> order_placed_on
- order_delivered_customer_date -> order_delivered_on

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 115609 entries, 0 to 115608
Data columns (total 40 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   customer_id                    115609 non-null  object 
 1   order_id                       115609 non-null  object 
 2   order_item_id                  115609 non-null  int64  
 3   customer_unique_id             115609 non-null  object 
 4   customer_state                 115609 non-null  object 
 5   customer_city                  115609 non-null  object 
 6   product_id                     115609 non-null  object 
 7   product_category_name          115609 non-null  object 
 8   price                          115609 non-null  float64
 9   product_name_lenght            115609 non-null  float64
 10  product_description_lenght     115609 non-null  float64
 11  product_photos_qty             115609 non-null  float64
 12  product_weight_g              

###  Handling Multiple Rows for a Single Order

- All rows belonging to the same `order_id` were grouped together.  
- For each order:
  - The `payment_value` values were **summed** to get the total amount paid.
  - The `payment_sequential` was replaced with its **maximum value** to represent the total number of transactions made.
  - The `payment_type` was marked as the actual payment type if all transactions used the same method, otherwise labeled as **"multiple"**.
- As a result, every order now exists as a **single row**, representing complete payment information in one record instead of having multiple rows.


### 📦 Transforming `order_item_id` into `items_in_order`
After converting multple rows of one order into **one single row**   
keeping multiple `order_item_id` values was no longer meaningful.

- All rows corresponding to the same `order_id` were grouped together.
- The maximum value of `order_item_id` will be stored in a new column **`items_in_order`** .

**Outcome:**  
The dataset now includes a single intuitive feature — `items_in_order` — indicating the total number of products per order.  


In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 115609 entries, 0 to 115608
Data columns (total 40 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   customer_id                    115609 non-null  object 
 1   order_id                       115609 non-null  object 
 2   order_item_id                  115609 non-null  int64  
 3   customer_unique_id             115609 non-null  object 
 4   customer_state                 115609 non-null  object 
 5   customer_city                  115609 non-null  object 
 6   product_id                     115609 non-null  object 
 7   product_category_name          115609 non-null  object 
 8   price                          115609 non-null  float64
 9   product_name_lenght            115609 non-null  float64
 10  product_description_lenght     115609 non-null  float64
 11  product_photos_qty             115609 non-null  float64
 12  product_weight_g              

### ⏱️ Delivery Time Calculation

- we will create a new column new **`delivery_time`**
- by using  `order_delivered_customer_date` and `order_purchase_timestamp`.  
- considering only the **days** (ignoring hours, minutes, and seconds).  
- After creating this column, delete previous columns

In [20]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 115609 entries, 0 to 115608
Data columns (total 40 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   customer_id                    115609 non-null  object 
 1   order_id                       115609 non-null  object 
 2   order_item_id                  115609 non-null  int64  
 3   customer_unique_id             115609 non-null  object 
 4   customer_state                 115609 non-null  object 
 5   customer_city                  115609 non-null  object 
 6   product_id                     115609 non-null  object 
 7   product_category_name          115609 non-null  object 
 8   price                          115609 non-null  float64
 9   product_name_lenght            115609 non-null  float64
 10  product_description_lenght     115609 non-null  float64
 11  product_photos_qty             115609 non-null  float64
 12  product_weight_g              

### 📦 Order Volume Calculation

-  delivery charge depends on the volume of a product.
-  create a new column **`order_volume`** .  
-  by combining `product_length_cm`, `product_height_cm`, `product_width_cm`, and `product_weight_g` .
- divide by a reasonle factor to make it suitable for analysis.
- now delete old columns

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 115609 entries, 0 to 115608
Data columns (total 40 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   customer_id                    115609 non-null  object 
 1   order_id                       115609 non-null  object 
 2   order_item_id                  115609 non-null  int64  
 3   customer_unique_id             115609 non-null  object 
 4   customer_state                 115609 non-null  object 
 5   customer_city                  115609 non-null  object 
 6   product_id                     115609 non-null  object 
 7   product_category_name          115609 non-null  object 
 8   price                          115609 non-null  float64
 9   product_name_lenght            115609 non-null  float64
 10  product_description_lenght     115609 non-null  float64
 11  product_photos_qty             115609 non-null  float64
 12  product_weight_g              

In [50]:
df.isnull().sum()

customer_id                           0
order_id                              0
order_item_id                         0
customer_unique_id                    0
customer_state                        0
customer_city                         0
product_id                            0
product_category_name                 0
price                                 0
product_name_lenght                   0
product_description_lenght            0
product_photos_qty                    0
product_weight_g                      1
product_length_cm                     1
product_height_cm                     1
product_width_cm                      1
product_category_name_english         0
seller_id                             0
seller_state                          0
seller_city                           0
seller_zip_code_prefix                0
review_id                             0
review_score                          0
review_comment_title             101808
review_comment_message            66703
review_creation_date                  0
review_answer_timestamp               0
order_purchase_timestamp              0
order_approved_at                    14
order_delivered_carrier_date       1195
order_delivered_customer_date      2400
order_status                          0
order_estimated_delivery_date         0
payment_type                          0
payment_installments                  0
payment_value                         0
payment_sequential                    0
freight_value                         0
customer_zip_code_prefix              0
shipping_limit_date                   0
dtype: int64

In [ ]:
# STATE COLUMNS

In [29]:
df['seller_state'].value_counts()
df['customer_state'].value_counts()

customer_state
SP    48797
RJ    14987
MG    13429
RS     6413
PR     5879
SC     4218
BA     3942
DF     2449
GO     2359
ES     2300
PE     1851
CE     1527
MT     1106
PA     1081
MS      845
MA      832
PB      619
PI      561
RN      560
AL      455
SE      393
TO      333
RO      279
AM      168
AC       93
AP       83
RR       50
Name: count, dtype: int64

###  State Category Consolidation

- Both `customer_state` and `seller_state` contained many low-frequency categories, with a heavy dominance of "SP" (São Paulo).  
-  all states with fewer than 1000 records were grouped together under a new category called **"Other"**.  


In [ ]:
# ID COLUMNS

In [37]:
print(df['order_id'].nunique())
print(df['customer_unique_id'].nunique())
print(df['seller_id'].nunique())
print(df['product_id'].nunique())



96516
93396
3028
32171


In [41]:
# PRODUCT CATEGORY
df['product_category_name_english'].unique()

array(['housewares', 'perfumery', 'auto', 'pet_shop', 'stationery',
       'furniture_decor', 'office_furniture', 'garden_tools',
       'computers_accessories', 'bed_bath_table', 'toys', 'telephony',
       'health_beauty', 'electronics', 'baby', 'cool_stuff',
       'watches_gifts', 'air_conditioning', 'sports_leisure',
       'books_general_interest', 'construction_tools_construction',
       'small_appliances', 'food', 'luggage_accessories',
       'fashion_underwear_beach', 'christmas_supplies',
       'fashion_bags_accessories', 'musical_instruments',
       'construction_tools_lights', 'books_technical',
       'costruction_tools_garden', 'home_appliances', 'market_place',
       'agro_industry_and_commerce', 'party_supplies', 'home_confort',
       'cds_dvds_musicals', 'industry_commerce_and_business',
       'consoles_games', 'furniture_bedroom', 'construction_tools_safety',
       'fixed_telephony', 'drinks',
       'kitchen_dining_laundry_garden_furniture', 'fashion_shoes',


**categorized the product category**

Example -
home appliance, home infrastructure, home_confort, home_appliances_2 -> home

In [44]:
# REVIEW SCORE
print(df['review_score'].value_counts())
print(df['review_score'].unique())

# all good here

review_score
5    65374
4    21951
1    14546
3     9718
2     4020
Name: count, dtype: int64
[4 5 1 2 3]


In [58]:
# DELIVERY RELATED COLUMNS

# check data type
print(df['order_purchase_timestamp'].dtype)
print(df['order_delivered_customer_date'].dtype)

#check missing val
print(df['order_purchase_timestamp'].isnull().sum())
print(df['order_delivered_customer_date'].isnull().sum())

object
object
0
2400


- need to change their data type
- need to handle missing value in delivered to customer column

In [59]:
# ORDER STATUS
df['order_status'].value_counts()

order_status
delivered      113210
shipped          1138
canceled          536
invoiced          358
processing        357
unavailable         7
approved            3
Name: count, dtype: int64

### MAKE ONLY 3 CATEGORY
- delivered
- processing -> shipped + invoiced + processing + approved
- canceled -> canceled + unavailable

In [60]:
# PAYMENT TYPE
df['payment_type'].value_counts()

payment_type
credit_card    85278
boleto         22510
voucher         6162
debit_card      1659
Name: count, dtype: int64

- boleto is also a voucher type payment in cash format
- merge it into the voucher type

In [69]:
# PAYMENT INSTALLMENT
print(df['payment_installments'].value_counts())

# check payment installment = 0
df[df['payment_installments'] == 0]


payment_installments
1     57599
2     13404
3     11551
4      7855
10     6785
5      5928
8      5013
6      4546
7      1789
9       710
12      164
15       91
18       38
24       34
11       22
20       20
13       19
14       16
16        7
17        7
21        6
0         3
23        1
22        1
Name: count, dtype: int64


np.int64(0)

- for 73979 -> manually edit 1
- for 77063 & 77064 both have same order -> manually edit 2

In [72]:
# PAYMENT VALUE
df['payment_value'].describe()

#check for 0

df[df['payment_value'] == 0]


customer_id                          order_id  \
491     8eab8f9b3c744b76b65f7a2c0c8f2d6c  45ed6e85398a87c253db47c2d9f48216   
30202   9af2372a1e49340278e7c1ef8d749f34  fa65dad1b0e818e3ccc5cb0e39231352   
30211   9af2372a1e49340278e7c1ef8d749f34  fa65dad1b0e818e3ccc5cb0e39231352   
37659   843b211abe7b0264dd4a69eafc5bdf43  6ccb433e00daae1283ccc956189c82ae   
99640   648121b599d98c420ef93f6135f8c80c  b23878b3e8eb4d25a158f57d96331b18   
112728  f2def7f64f36952f2f5a9791f0285f34  8bcbe01d44d147f901cd3192671144db   

        order_item_id                customer_unique_id customer_state  \
491                 1  e4000306cf2f63714e6bb70dd20a6592             GO   
30202               1  8af7ac63b2efbcbd88e5b11505e8098a             MT   
30211               1  8af7ac63b2efbcbd88e5b11505e8098a             MT   
37659               1  677ad9cdca6c47c733f0cc6e23d7bb4c             RJ   
99640               1  9077449283dc1319d5e51fb3159b28e2             ES   
112728              1  cc5aa1b8337e394da54980226bb0f6d4             RJ   

         customer_city                        product_id  \
491            goiania  75d6b6963340c6063f7f4cfcccfe6a30   
30202           cuiaba  1aecdb5fa3add74e385f25c6c527a462   
30211           cuiaba  1aecdb5fa3add74e385f25c6c527a462   
37659       petropolis  2ee4be9805f228255a38a6f9b03fff1a   
99640         anchieta  e306ca54c91b21392317d5b4632c9fe3   
112728  rio de janeiro  85d4c1a46f08f730de651ea6f6645313   

       product_category_name   price  product_name_lenght  ...  \
491               perfumaria   56.99                 51.0  ...   
30202     ferramentas_jardim  392.55                 41.0  ...   
30211     ferramentas_jardim  392.55                 41.0  ...   
37659          esporte_lazer   94.00                 38.0  ...   
99640         consoles_games  135.30                 42.0  ...   
112728             papelaria   59.00                 50.0  ...   

        order_delivered_customer_date  order_status  \
491               2017-06-22 17:44:53     delivered   
30202                             NaN       shipped   
30211                             NaN       shipped   
37659             2017-11-08 18:47:00     delivered   
99640             2017-06-07 08:53:10     delivered   
112728            2018-02-02 21:26:42     delivered   

        order_estimated_delivery_date  payment_type  payment_installments  \
491               2017-07-10 00:00:00       voucher                     1   
30202             2017-05-18 00:00:00       voucher                     1   
30211             2017-05-18 00:00:00       voucher                     1   
37659             2017-11-22 00:00:00       voucher                     1   
99640             2017-06-29 00:00:00       voucher                     1   
112728            2018-02-21 00:00:00       voucher                     1   

        payment_value payment_sequential freight_value  \
491               0.0                  3         14.15   
30202             0.0                 14         65.44   
30211             0.0                 13         65.44   
37659             0.0                  4         28.04   
99640             0.0                  4         36.27   
112728            0.0                  4         15.16   

       customer_zip_code_prefix  shipping_limit_date  
491                       74595  2017-06-15 21:30:18  
30202                     78065  2017-04-27 09:10:13  
30211                     78065  2017-04-27 09:10:13  
37659                     25625  2017-11-03 00:46:58  
99640                     29230  2017-06-01 19:35:17  
112728                    21040  2018-01-30 23:37:20  

[6 rows x 40 columns]

- for 491 -> payment installment & sequential available so manually edit the payment value
- for 30202 & 30211 -> it is a single order from 30194 to 30222 so merge all column it will automatically give payment value
- same for 37659 -> 37657 to 37662 same order , merge them
- same for 99640
- same for 112728

In [76]:
# PAYMENT SEQUENTIAL
df['payment_sequential'].value_counts()


# all good

payment_sequential
1     110662
2       3298
3        636
4        307
5        183
6        127
7         88
8         58
9         47
10        40
11        35
12        27
13        15
14        13
15        11
18         9
17         9
16         9
19         9
20         6
21         6
22         3
25         2
23         2
26         2
24         2
27         1
29         1
28         1
Name: count, dtype: int64

In [84]:
# FREIGHT VALUE
print(df['freight_value'].describe())


# check for max
df[df['freight_value'] == 409.68]

# valid max value

count    115609.000000
mean         20.056880
std          15.836184
min           0.000000
25%          13.080000
50%          16.320000
75%          21.210000
max         409.680000
Name: freight_value, dtype: float64


customer_id                          order_id  \
58611  4eed76c1aafeceac4250e202ca9b1a5a  a77e1550db865202c56b19ddc6dc4d53   

       order_item_id                customer_unique_id customer_state  \
58611              1  97734fdca127fddcb5f92f841690c3f2             PI   

             customer_city                        product_id  \
58611  sao raimundo nonato  ec31d2a17b299511e7c8627be9337b9b   

      product_category_name  price  product_name_lenght  ...  \
58611                 bebes  979.0                 44.0  ...   

       order_delivered_customer_date  order_status  \
58611            2018-04-27 19:45:00     delivered   

       order_estimated_delivery_date  payment_type  payment_installments  \
58611            2018-05-17 00:00:00   credit_card                    12   

       payment_value payment_sequential freight_value  \
58611        1388.68                  1        409.68   

      customer_zip_code_prefix  shipping_limit_date  
58611                    64770  2018-04-25 06:55:13  

[1 rows x 40 columns]

In [85]:
#check for 0
df[df['freight_value'] == 0]

customer_id                          order_id  \
233     9e5ce657315b2bdb94033a494041ac25  6d7de866a140b19d09e825b2a4e944c7   
234     9e5ce657315b2bdb94033a494041ac25  6d7de866a140b19d09e825b2a4e944c7   
482     ad4da04c9e20699c6ad1843af39846d2  d62387acb3914789e712406e9882e549   
752     55216cd56c9eaadec16f03d5aaf11d86  f29d0fd6d4e6d5ce550e0b2f9335116c   
775     fdabedbba76cf0308dedcd99afc66626  a2ea081c06c12c25c068089e4985745a   
...                                  ...                               ...   
114309  68b8c86408d7f6fe483107fa73f4dca4  f937ec3a717fd77710c5fb85cf5321d9   
114509  dc5e8db48a49602f506e67b843668d72  577b745a85120419fe27073e3c9e1a76   
114774  a7dd8bd4bdbdc55196a8ee7fd336dde7  b811c8ca07716586e80c0bc4f2a6e39f   
114832  d5bb85d2c9324fc7a298992e501ca509  f565be42bf5390f6cd98934cc4c17091   
115230  a7e51ce898bb8caad5301b006e61bbb9  a835cb24fb9dc47511a9932df13180cf   

        order_item_id                customer_unique_id customer_state  \
233                 1  51cbfa44126505de7a55fb99ba49648b             SC   
234                 2  51cbfa44126505de7a55fb99ba49648b             SC   
482                 1  1f7d9afd77ceacfa0b5b461a50818806             RJ   
752                 1  c1dc1162de49817a25314db3030276f5             RS   
775                 1  826303f32007dc49ba0fbf597caf80ec             CE   
...               ...                               ...            ...   
114309              1  6148c0fb32e5c7352da8656b9b34ec60             RJ   
114509              1  e21bea57967ffc7bdc8b07558e09685e             RJ   
114774              1  a9747b164275fabdd901a9fc169c5251             RJ   
114832              1  f3a6febe6e4236817f26caf440beea4e             SP   
115230              1  934d5b3c394e8ecf4c1124e1b0af7e82             RJ   

         customer_city                        product_id  \
233            palhoca  aca2eb7d00ea1a7b8ebd4e68314663af   
234            palhoca  aca2eb7d00ea1a7b8ebd4e68314663af   
482     rio de janeiro  53b36df67ebb7c41585e8d54d6772e08   
752      campos borges  aca2eb7d00ea1a7b8ebd4e68314663af   
775            massape  53b36df67ebb7c41585e8d54d6772e08   
...                ...                               ...   
114309         niteroi  53b36df67ebb7c41585e8d54d6772e08   
114509  rio de janeiro  7a10781637204d8d10485c71a6108a2e   
114774    belford roxo  aca2eb7d00ea1a7b8ebd4e68314663af   
114832          araras  53b36df67ebb7c41585e8d54d6772e08   
115230     sao goncalo  53b36df67ebb7c41585e8d54d6772e08   

       product_category_name  price  product_name_lenght  ...  \
233         moveis_decoracao   69.9                 44.0  ...   
234         moveis_decoracao   69.9                 44.0  ...   
482       relogios_presentes   99.9                 33.0  ...   
752         moveis_decoracao   69.9                 44.0  ...   
775       relogios_presentes  110.0                 33.0  ...   
...                      ...    ...                  ...  ...   
114309    relogios_presentes   99.9                 33.0  ...   
114509    relogios_presentes  199.0                 42.0  ...   
114774      moveis_decoracao   69.9                 44.0  ...   
114832    relogios_presentes   99.9                 33.0  ...   
115230    relogios_presentes  106.9                 33.0  ...   

        order_delivered_customer_date  order_status  \
233               2018-04-24 20:38:55     delivered   
234               2018-04-24 20:38:55     delivered   
482               2018-05-14 19:05:08     delivered   
752               2018-06-09 00:14:23     delivered   
775               2018-05-26 00:08:34     delivered   
...                               ...           ...   
114309            2018-05-14 16:46:35     delivered   
114509            2018-05-07 13:22:01     delivered   
114774            2018-05-01 00:32:42     delivered   
114832            2018-05-25 23:51:46     delivered   
115230            2018-05-11 18:17:45     delivered   

        order_estimated_delivery_date  p

### for charge = o
- some have right values like no delivery charge taken
- some have problem like order value = 70  & payment value = 120 & charges = 0 , so this is not logically possible and for this kind of issue we have to manually edit the charges(freight) values.

### how to check

- payment value = order value + frieght value